### DISCUSSION TITLE
Filter group

### DISCUSSION THREAD

**Discussion by ViachaslauBusel** [bepu/bepuphysics2#176] [Q&A] [answered]
How to create a filter that is used to selectively ignore objects when casting a ray. I want to split the objects in the simulation into two groups, and selectively ignore one of the groups when casting a ray. thanks

**Comment by RossNordby** [comment, marked as answer]
`IRayHitHandler` has an `AllowTest` callback that is called before doing any testing. It passes in a collidable reference you can use to look up filter data.

Exactly how the filtering data is stored or computed is up to you. It can be helpful to have a mapping from collidable reference to other data- `CollidableProperty<T>` is a helper class that exists for that purpose. For example, to implement narrow phase pair filtering, the ragdoll demo uses it: https://github.com/bepu/bepuphysics2/blob/11f7e09888759f889fe6e2aa687a1bd74627a735/Demos/Demos/RagdollDemo.cs#L97.
